hs_app_timeseries/forms.py

```diff
@@ -143,38 +245,104 @@ class ProcessingLevelValidationForm(forms.Form):
+        self.fields['sample_medium'].widget = forms.Select(choices=cv_sample_medium_choices)
+        cv_units_type_choices = _get_cv_dropdown_widget_items(self.cv_units_types,
+                                                              current_units_type)
+        self.fields['units_type'].widget = forms.Select(choices=cv_units_type_choices)
+        cv_status_choices = _get_cv_dropdown_widget_items(self.cv_statuses, current_status)
+        self.fields['status'].widget = forms.Select(choices=cv_status_choices)
+        cv_agg_statistics_choices = _get_cv_dropdown_widget_items(self.cv_aggregation_statistics,
+                                                                  current_agg_statistics)
+        self.fields['aggregation_statistics'].widget = forms.Select(
+            choices=cv_agg_statistics_choices)
+
+    @property
+    def form_id(self):
+        form_id = 'id_timeseriesresult_%s' % self.number
+        return form_id
+
+    @property
+    def form_id_button(self):
+        form_id = 'id_timeseriesresult_%s' % self.number
+        return "'" + form_id + "'"
 
     class Meta:
         model = TimeSeriesResult
-        fields = ['units_type', 'units_name', 'units_abbreviation', 'status', 'sample_medium', 'value_count',
-                  'aggregation_statistics']
+        fields = ['units_type', 'units_name', 'units_abbreviation', 'status', 'sample_medium',
+                  'value_count', 'aggregation_statistics']
         widgets = {'value_count': forms.TextInput()}
 
+
 class TimeSeriesResultValidationForm(forms.Form):
     units_type = forms.CharField(max_length=255)
     units_name = forms.CharField(max_length=255)
     units_abbreviation = forms.CharField(max_length=20)
     status = forms.CharField(max_length=255)
     sample_medium = forms.CharField(max_length=255)
     value_count = forms.IntegerField()
-    aggregation_statistics = forms.CharField(max_length=255)
\ No newline at end of file
+    aggregation_statistics = forms.CharField(max_length=255)
+
+
+def _get_cv_dropdown_widget_items(dropdown_items, selected_item_name):
+    for item in dropdown_items:
+        if item.name == selected_item_name:
+            dropdown_items.remove(item)
+
+    cv_items = [(item.name, item.name) for item in dropdown_items]
+    cv_items = [(selected_item_name, selected_item_name)] + cv_items
+    cv_item_choices = tuple(cv_items)
+    return cv_item_choices
+
+
+def _get_html_snippet(html_snippet_file_name):
+    cur_dir = os.path.dirname(os.path.realpath(__file__))
+    html_file = os.path.join(cur_dir, 'layout_snippets', html_snippet_file_name)
+    with open(html_file, 'r') as fobj:
+        html_snippet = fobj.read()
+    return html_snippet
+
+UpdateSQLiteLayout = Layout(HTML(_get_html_snippet('update_sqlite_file.html')))
+SeriesSelectionLayout = Layout(HTML(_get_html_snippet('series_selection.html')))
+TimeSeriesMetaDataLayout = HTML(_get_html_snippet('timeseries_metadata_edit.html'))
```

Comment:
@pkdash I am wondering if these three layout snippets are only used here for layout, perhaps it is better to define html snippets as multi-line strings to be used directly in the code to create layouts here rather than reading them from html files, just from performance standpoint.